### PR TITLE
Editor: Fix `remove_control_from_dock` fails when dock is floating.

### LINKED ISSUE #78878: Crash "Control was not in dock." (edgecase)
### Godot version

v4.1.rc2.official [46424488e]

### System information

Godot v4.1.rc2 - macOS 13.4.1 - Vulkan (Compatibility) - Apple M1 - Apple M1 (8 Threads)

### Issue description

```
Reloading GD OpenAI coding.
ERROR: Control was not in dock.
   at: remove_control_from_dock (editor/editor_node.cpp:5835)
(empty)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.1.rc2.official (46424488edc341b65467ee7fd3ac423e4d49ad34)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] 1   libsystem_platform.dylib            0x0000000181aaaa24 _sigtramp + 56
[2] AspectRatioContainer::get_alignment_vertical() const
[3] ShaderGlobalsEditorInterface::_get_property_list(List<PropertyInfo, DefaultAllocator>*) const
[4] JSON::get_data() const
[5] Object::get_instance_id() const
[6] Ref<DirAccess> DirAccess::_create_builtin<DirAccessMacOS>()
[7] RendererCompositorRD::_create_current()
[8] 8   AppKit                              0x0000000185044f2c __19-[NSWindow __close]_block_invoke + 136
[9] 9   AppKit                              0x0000000185044e90 -[NSWindow __close] + 272
[10] 10  AppKit                              0x0000000184f218a8 -[NSApplication(NSResponder) sendAction:to:from:] + 440
[11] 11  AppKit                              0x0000000184f216c0 -[NSControl sendAction:to:] + 72
[12] 12  AppKit                              0x0000000184f21604 __26-[NSCell _sendActionFrom:]_block_invoke + 100
[13] 13  AppKit                              0x0000000184f2152c -[NSCell _sendActionFrom:] + 204
[14] 14  AppKit                              0x0000000184f21450 -[NSButtonCell _sendActionFrom:] + 88
[15] 15  AppKit                              0x0000000184f1ea54 NSControlTrackMouse + 1480
[16] 16  AppKit                              0x0000000184f1e460 -[NSCell trackMouse:inRect:ofView:untilMouseUp:] + 144
[17] 17  AppKit                              0x0000000184f1e318 -[NSButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 488
[18] 18  AppKit                              0x0000000184f1d7e4 -[NSControl mouseDown:] + 448
[19] 19  AppKit                              0x0000000184f1c2cc -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 3476
[20] 20  AppKit                              0x0000000184ea6f08 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 364
[21] 21  AppKit                              0x0000000184ea6bc8 -[NSWindow(NSEventRouting) sendEvent:] + 284
[22] 22  AppKit                              0x0000000184ea5f0c -[NSApplication(NSEvent) sendEvent:] + 1556
[23] Ref<DirAccess> DirAccess::_create_builtin<DirAccessMacOS>()
[24] Ref<DirAccess> DirAccess::_create_builtin<DirAccessMacOS>()
[25] 25  Godot                               0x00000001051482c8 Godot + 4031176
[26] RendererCompositorRD::_create_current()
[27] 27  dyld                                0x0000000181723f28 start + 2236
-- END OF BACKTRACE --
================================================================
[1]    12342 abort      /Applications/Godot_v4.1-rc2_macos.universal.app/Contents/MacOS/Godot 
```

### Steps to reproduce

Allow editor plugin to reload itself and make it float. Then reload then close the floating empty 

## Plugin with reload button

image

## Reloaded plugin with empty float

image

## Script used

```gdscript
func _enter_tree():
	dock = preload("res://addons/gd-openai-coding/dock.tscn").instantiate()
	add_control_to_dock(DOCK_SLOT_RIGHT_BL, dock)

	dock.find_child("Reload").connect("pressed", self.reload)

func _exit_tree():
	remove_control_from_docks(dock)
	dock.queue_free()

func reload():
	print("Reloading GD OpenAI coding.")
	_exit_tree()
	_enter_tree()
```

### Minimal reproduction project

[test-gd-openai.zip](https://github.com/godotengine/godot/files/11917837/test-gd-openai.zip)

### PR DESCRIPTION
**Pull request by zaevi**
Fixes #78878.

### CODE REVIEW

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -5850,6 +5850,14 @@ void EditorNode::add_control_to_dock(DockSlot p_slot, Control *p_control) {
 }
 
 void EditorNode::remove_control_from_dock(Control *p_control) {
+	// If the dock is floating, close it first.
+	for (WindowWrapper *wrapper : floating_docks) {
+		if (p_control == wrapper->get_wrapped_control()) {
+			wrapper->set_window_enabled(false);
```
Comment (zaevi) [inline editor/editor_node.cpp:5856]: `set_window_enabled(false)` will emit `window_close_requested` signal and the control will be back to editor's dock.

**Comment by zaevi** [comment]
A quick reproduction for #78878:

```gdscript
@tool
extends EditorPlugin

var btn : Button

func _enter_tree():
	btn = Button.new()
	add_control_to_dock(EditorPlugin.DOCK_SLOT_LEFT_BL, btn)

func _exit_tree():
	if btn:
		remove_control_from_docks(btn) # 
		btn.queue_free()
		btn = null
```

1. Enable the plugin and make it floating.
2. Disable the plugin. `remove_control_from_docks` will fail and the floating window will stay. (this PR fixes)
3. Close the floating window, causes crash.

**Comment by kitbdev** [comment]
Works as expected and code looks good.

**Comment by AThousandShips** [comment]
I believe this matches roughly what I attempted for this, will do some testing of this when I'm able

**Review by KoBeWi — APPROVED** [review]
CC @trollodel

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**